picasso-compose/src/main/java/com/squareup/picasso3/compose/PicassoPainter.kt

```diff
@@ -100,6 +117,20 @@ internal class PicassoPainter(
     errorDrawable?.let(::setPainter)
   }
 
+  private fun load() {
+    // This derived state read will return the same instance of RequestCreator if one has been
+    // cached and none of the state dependencies have since changed.
+    val requestCreator = requestCreator.value
+    // lastRequestCreator is just used for diffing, we don't want it to invalidate anything.
+    val lastRequestCreator = Snapshot.withoutReadObservation { lastRequestCreator }
+
+    // Only launch a new request if anything has actually changed.
+    if (requestCreator != lastRequestCreator) {
```

Comment:
I think `RequestCreator` doesn't implement `equals` so this is effectively just a reference check. Might be worth implementing `equals` to avoid even more re-requests. Might be worth implementing `equals` to make this more meaningful. Or at least documenting that limitation with a comment here.